src/membership/components/MemberForm.jsx

```diff
@@ -29,6 +29,43 @@ const PreviewBadgeButton = ({ member, style }) => (
   </PreviewBadge>
 )
 
+const BadgeRow = ({ getMsg, inputProps, isAdmin, member }) => (
+  <ConfigConsumer>
+    {({ getMemberAttr }) =>
+      getMemberAttr(member).badge ? (
+        <Row style={{ alignItems: 'flex-end' }}>
+          <Col xs={12} sm={6}>
+            <TextInput
+              hintText={member.get('preferred_name')}
+              multiLine
+              path="badge_name"
+              rowsMax={2}
+              {...inputProps}
+            />
+          </Col>
+          <Col xs={12} sm={isAdmin ? 3 : 6} md={isAdmin ? 4 : 6}>
+            <TextInput
+              {...inputProps}
+              path="badge_subtitle"
+              hintText={member.get('country')}
+            />
+          </Col>
+          {isAdmin ? (
+            <Col xs={12} sm={3} md={2}>
+              <PreviewBadgeButton member={member} style={{ float: 'right' }} />
+            </Col>
+          ) : (
+            <Col xs={12} style={hintStyle}>
+              <PreviewBadgeButton member={member} style={{ float: 'right' }} />
+              {getMsg('badge_hint')}
+            </Col>
+          )}
+        </Row>
+      ) : null
```

Comment:
Sneeky null :).

This conditional logic is more suited to single line styles, but I'm having a hard time tracing back to the conditions.

Could we use guarded returns and pull this to the top?

Something like...

```
badge = getMemberAttr(member).badge
if (!badge) return null
return (
   <Row ...
```

Reply:
I guess with React you get used to more of these, as you see and use more value statements. We're getting `getMemberAttr` from the context consumer, so it does need to be within a child. Something like `getMemberAttr(member).badge && (...)` would also work, as all falsy values mean that nothing is rendered for the component.